### ISSUE TITLE
[Bug]: Re-spawning in tent after becoming human again

### ISSUE THREAD

**Issue by MarioSMB** [TeamLapen/Vampirism#1303]
### Minecraft Version

1.20.x

### Forge Version

47.2.17

### Vampirism Version

1.10.0

### Singleplayer/Multiplayer

Server

### What happened?

When I cleansed myself of being a hunter, I respawned at a tent I'd set spawn at as a hunter!  
Perhaps it should mention that my spawn point is no longer valid or something?

**Comment by Cheaterpaul** [comment]
This is the correct behavior.

The tent is no different when it comes to setting the spawnpoint, resetting the phantom spawns or sleeping.

**Comment by MarioSMB** [comment]
It is expected that I can respawn at a tent when I am not a hunter?  

Admittedly this is a nitpicky detail and one that doesn't necessarily need to be addressed, but it would make more sense to tell the player their spawn point is no longer available in this situation.

**Comment by lunofe** [comment]
Deleting the spawn point just seems like an unnecessary burden on the player

**Comment by Cheaterpaul** [comment]
Checking the spawnpoint when switching the factions can cause loading of unloaded chunks, which is not good.

So this behavior is fine.

**Comment by MarioSMB** [comment]
That's strange - does vanilla chunkload your bed or respawn anchor area when checking to see if it was obstructed or destroyed?  
Surely there would be some caching or invalidation possible to avoid needing to do so, though as noted this is only a minor bug, so leaving it in may be fine.

**Comment by maxanier** [comment]
It would probably be possible to delay the check and potential reset of spawnpoint until the player actually has to respawn to avoid unnecessary chunk loading.
However, as lunofe said, this would be quite annoying. Do you think it breaks immersion so much, that this is worth it?

**Comment by MarioSMB** [comment]
Well... it was something I encountered that seemed like a very strange inconsistency/bug, given you can *only* use a hunter tent as a hunter.  
For extra amusement, you can respawn at a hunter tent as a vampire!  

It *is* a little annoying to lose your respawn point even in vanilla, but it's a rule of the game, just as the mod defines the rule that you cannot use a hunter tent unless you're a hunter - so it *should* be expected that you lose your hunter tent spawn point, just as you should expect to lose access to your coffin if you cleanse yourself from vampirism (along with the ability to use any exclusive items like swords and crossbows).   
\
\
As you say though, it may not be appealing to fix; bugs quite often become features if they make the experience better.  
It's a very minor detail and there are valid concerns of it being an annoyance more than anything (though I cannot say without testing), so there's certainly no pressing need to address it.

**Comment by Gregory47303** [comment]
Yeah I don't think this is worth wasting a single second on tbh, I wouldn't even describe it as a bug either it's a bloody respawn point it's not some crazy game breaking thing like vampires being able to use hunter pots.  

Just seems like something this bloke and this bloke alone is worried about tbh never once heard anyone even mention this nor actually care

**Comment by MarioSMB** [comment]
> Just seems like something this bloke and this bloke alone is worried about tbh never once heard anyone even mention this nor actually care

You know... I'm getting really tired of being targeted for reporting obscure issues.  
Just because nobody else reported an issue, that doesn't mean nobody has encountered it or that it's invalid.  

Let's keep this constructive, please.

**Comment by lunofe** [comment]
Look, Mario, I don't agree with the guy's tone either, but there is actual valid feedback in his message that shouldn't be dismissed as a personal attack. You mentioned it "should be expected behavior" and is just a minor annoyance, but you're looking at this strictly from your own perspective and missing other playstyles.

For context: in my vanilla world, I traveled 70,000 blocks over three nights just to find the perfect spot to build. Imagine playing Vampirism, losing your Hunter status, and having your spawn point wiped just because it feels "logical." Am I supposed to walk another 70,000 blocks just because it's technically consistent?

This is a classic game design trap. Games don't need to be perfectly logical—they need to be fun. If real-world logic ruins the player experience, it doesn't make sense to enforce it. That’s why people are criticizing this. The execution of the feedback was poor, but the core point stands.

**Comment by maxanier** [comment]
Please keep reporting issues, even if they are a bit obscure. We can still decide whether to "fix" them.
If they are minor, you may want to combine them into a single issue.

This specific issue won't be fixed for a) lunofe's point, b) it actually would get ugly in code.

Side note: You cannot respawn in tents at all in 1.21. Going to fix that.

**Comment by MarioSMB** [comment]
Thank you for approaching this with reason and consideration - it was more of a "huh, that's weird" encounter than some major issue.  

For the record, I don't think any proposed solution would involve punishing the player (i.e. only letting them know their spawnpoint was invalidated *when* they respawn). Vanilla does that, and as pointed out, it's really annoying.  
Issues like this help show room for improvement, not necessarily a need to "fix" them.  

I'm glad it helped catch a bug, if nothing else!
